directives/animate/test/animate.test.js

```diff
@@ -0,0 +1,107 @@
+import { expect, fixture, oneEvent } from '@open-wc/testing';
+import { forceFocusVisible, getComposedActiveElement } from '../../../helpers/focus.js';
+import { hide, show } from '../animate.js';
+import { html, LitElement } from 'lit-element/lit-element.js';
+
+class FocusTestElem extends LitElement {
+
+	static get properties() {
+		return {
+			animate: { type: Boolean }
+		};
+	}
+
+	render() {
+		const animateValue = this.animate ? hide() : undefined;
+		return html`
+			<button id="first" .animate="${animateValue}" @d2l-animate-complete="${this._dispatchEvent}">first</button>
+			<button id="second">second</button>
+		`;
+	}
+
+	focus() {
+		this.shadowRoot.querySelector('#first').focus();
+	}
+
+	forceFocusVisible() {
+		forceFocusVisible(this.shadowRoot.querySelector('#first'));
+	}
+
+	_dispatchEvent() {
+		this.dispatchEvent(new CustomEvent('d2l-animate-test-focus-animate-complete'));
+	}
+
+}
+customElements.define('d2l-animate-test-focus', FocusTestElem);
+
```

Comment:
I've added some tests in here to attempt to test some of the weirder edge cases that we'd like to not regress on in the future.